sdk/storage/azure-storage-file/src/main/java/com/azure/storage/file/FileServiceAsyncClient.java

```diff
@@ -157,20 +158,22 @@ public ShareAsyncClient getShareAsyncClient(String shareName, String snapshot) {
      * <a href="https://docs.microsoft.com/en-us/rest/api/storageservices/list-shares">Azure Docs</a>.</p>
      *
      * @param options Options for listing shares
+     * @param timeout An optional timeout to be applied to the network asynchronous operations.
      * @return {@link ShareItem Shares} in the storage account that satisfy the filter requirements
      */
-    public PagedFlux<ShareItem> listShares(ListSharesOptions options) {
-        return listShares(null, options);
+    public PagedFlux<ShareItem> listShares(ListSharesOptions options, Duration timeout) {
```

Comment:
Same as the other async classes, also we need to add Context to the listing methods.